File: tests/browser/features/support/pages/special_version.rb
# Page containing MediaWiki version, plugin version, and version of external
# software like Elasticsearch.
class SpecialVersion
  include PageObject

  page_url URL.url("/wiki/Special:Version")

  table(:software_table, id: "sv-software")

  def software_table_row(name)
    software_table_element.cell_element(text: name)
  end
end

File: tests/integration/features/support/pages/special_version.js
/*jshint esversion: 6, node:true */

const TitlePage = require('./title_page');

class SpecialVersion extends TitlePage {
	constructor( title ) {
		super( title );
	}

	software_table_row( name ) {
		return this.table('#sv-software').element( 'td=' + name ).value;
	}
}

module.exports = new SpecialVersion( 'Special:Version' );
